test/integration/user-management-api/applications_test.rb

```diff
@@ -1,299 +1,287 @@
-    User.any_instance.stubs(:has_access_to_all_services?).returns(false)
-    user  = FactoryBot.create(:member, account: @provider, admin_sections: ['partners'])
-    token = FactoryBot.create(:access_token, owner: user, scopes: 'account_management')
-
-    get(admin_api_applications_path)
-    assert_response :forbidden
-    get(admin_api_applications_path, access_token: token.value)
-    assert_response :success
-    User.any_instance.expects(:member_permission_service_ids).returns([@service.id]).at_least_once
-    get(admin_api_applications_path, access_token: token.value, service_id: @service.id)
-    assert_response :success
-  end
-
-  # Provider key
-
-  test 'index on backend v2' do
-    @service.backend_version = '2'
-    @service.save!
-    get admin_api_applications_path(:format => :xml), :provider_key => @provider.api_key
-
-    assert_response :success
-    assert_applications @response.body, :backend => "2"
-  end
-
-  test 'index on backend v1' do
-    @service.backend_version = '1'
-    @service.save!
+  class AccessTokenTest < EnterpriseApiApplicationsTest
+    def setup
+      super
+      user = FactoryBot.create(:member, account: @provider, admin_sections: ['partners'])
+      @token = FactoryBot.create(:access_token, owner: user, scopes: 'account_management')
+    end
 
-    get admin_api_applications_path(:format => :xml), :provider_key => @provider.api_key
+    context 'with access to all services' do
+      setup do
+        User.any_instance.stubs(:has_access_to_all_services?).returns(true)
+      end
+
+      should '#index without token' do
+        get admin_api_applications_path
+        assert_response :forbidden
+      end
+
+      should '#index' do
+        User.any_instance.expects(:member_permission_service_ids).never
+        get admin_api_applications_path, params: params.merge({ service_id: @service.id })
+        assert_response :success
+      end
+    end
 
-    assert_response :success
-    assert_applications @response.body, :backend => "1"
-  end
+    context 'without access to all services' do
+      setup do
+        User.any_instance.stubs(:has_access_to_all_services?).returns(false)
+      end
 
-  test 'index on backend oauth' do
-    @service.backend_version = 'oauth'
-    @service.save!
+      should '#index without token' do
+        get admin_api_applications_path
+        assert_response :forbidden
+      end
 
-    get admin_api_applications_path(:format => :xml), :provider_key => @provider.api_key
+      should '#index with access to some service' do
+        User.any_instance.expects(:member_permission_service_ids).returns([@service.id]).at_least_once
+        get admin_api_applications_path, params: params.merge({ service_id: @service.id })
 
-    assert_response :success
-    assert_applications @response.body, :backend => :oauth
-  end
+        assert_response :success
+        assert_equal 2, Nokogiri::XML::Document.parse(@response.body).xpath("//applications").children.length
```

Comment:
Does this mean two applications? Where are 2 coming from?

Reply:
It comes from the first setup https://github.com/3scale/porta/pull/2710/files#diff-f0116a6712202a02588a528293a86589c4898120efb2ca2e64fd22399dc26299R9-R25

Buyer buys two times ;)

Reply:
Since this test is failing, I'll remove the hardcode number